src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -1738,5 +1740,39 @@ internal ArgPreprocessor GetGenericArgPreprocessor(int index)
 
             return ArgPreprocessor.None;
         }
+
+        /// <summary>
+        /// Visit all function nodes to compose dependency info.
+        /// </summary>
+        /// <param name="node">IR CallNode.</param>
+        /// <param name="visitor">Dependency visitor.</param>
+        /// <param name="context">Dependency context.</param>
+        /// <returns>Static boolean value.</returns>
+        public virtual bool ComposeDependencyInfo(IRCallNode node, DependencyVisitor visitor, DependencyContext context)
+        {
+            for (int i = 0; i < node.Args.Count; i++)
+            {
+                if (node.Scope != null && i < node.Function.ScopeArgs)
+                {
+                    if (node.Args[i] is IRCallNode callNode)
+                    {
+                        callNode.Accept(visitor, context);
+                    }
+                    else
+                    {
+                        continue;
+                    }
+                }
+                else
+                {
+                    node.Args[i].Accept(visitor, context);
+                }
+            }
+
+            // The return value is used by DepedencyScanFunctionTests test case.
+            // Returning false to indicate that the function runs a basic dependency scan.
+            // Other functions can override this method to return true if they have a custom dependency scan.
+            return false;

```

Comment:
What do these semantics mean?  Ie, what is a caller supposed to do differently if we return true vs. false?

Reply:
Discussed internally.